src/NumericalAlgorithms/Interpolation/InterpolationTargetLineSegment.hpp

```diff
@@ -0,0 +1,127 @@
+
+  LineSegment() = default;
+  LineSegment(const LineSegment& /*rhs*/) = default;
+  LineSegment& operator=(const LineSegment& /*rhs*/) = default;
+  LineSegment(LineSegment&& /*rhs*/) noexcept = default;
+  LineSegment& operator=(LineSegment&& /*rhs*/) noexcept = default;
+  ~LineSegment() = default;
+
+  // Need pupper so that this class can be used as a CacheTag.
+  // clang-tidy non-const reference pointer.
+  void pup(PUP::er& p) noexcept {  // NOLINT
+    p | begin;
+    p | end;
+    p | number_of_points;
+  }
+
+  std::array<double, VolumeDim> begin{};
+  std::array<double, VolumeDim> end{};
+  size_t number_of_points{};
+};
+}  // namespace OptionHolders
+
+namespace Actions {
+/// \ingroup ActionsGroup
+/// \brief Sends points on a line segment to an `Interpolator`.
+///
+/// Uses:
+/// - DataBox:
+///   - `::Tags::Domain<`VolumeDim`, Frame>`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// DataBox changes:
+/// - Adds: nothing
+/// - Removes: nothing
+/// - Modifies:
+///   - `Tags::IndicesOfFilledInterpPoints`
+///   - `::Tags::Variables<typename
+///                   InterpolationTargetTag::vars_to_interpolate_to_target>`
+///
+/// For requirements on InterpolationTargetTag, see InterpolationTarget
+template <typename InterpolationTargetTag, size_t VolumeDim, typename Frame>
+struct LineSegment {
+  using options_type = OptionHolders::LineSegment<VolumeDim>;
+  using const_global_cache_tags = tmpl::list<InterpolationTargetTag>;
+  template <typename DbTags, typename... InboxTags, typename Metavariables,
+            typename ArrayIndex, typename ActionList,
+            typename ParallelComponent,
+            Requires<tmpl::list_contains_v<
+                DbTags, typename Tags::TemporalIds<Metavariables>>> = nullptr>
+  static void apply(
+      db::DataBox<DbTags>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      Parallel::ConstGlobalCache<Metavariables>& cache,
+      const ArrayIndex& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/,
+      const typename Metavariables::temporal_id& temporal_id) noexcept {
+    const auto& options = Parallel::get<InterpolationTargetTag>(cache);
+
+    // Fill points on a line segment
+    const double dx = 1.0 / (options.number_of_points - 1);
+    tnsr::I<DataVector, VolumeDim, Frame> target_points(
+        options.number_of_points);
+    for (size_t n = 0; n < options.number_of_points; ++n) {
+      for (size_t d = 0; d < VolumeDim; ++d) {
+        target_points.get(d)[n] =
+            gsl::at(options.begin, d) +
+            n * dx * (gsl::at(options.end, d) - gsl::at(options.begin, d));
+      }
+    }
+
+    // All actions that can be chosen as
+    // InterpolateeTag::compute_target_points should fill
+    // target_points and then call the following function.
+    send_points_to_interpolator<InterpolationTargetTag>(
```

Comment:
I find this comment confusing, more than helpful.  Is there (or is there going to be) documentation describing how everything fits together where this comment would be more appropriate?

Reply:
Ok, removing this comment.  I just wanted to make it clear that all other compute_target_points functions that we write (Plane, Strahlkorper, etc.) fill their points and call send_points_to_interpolator.